PyPDF2/_writer.py

```diff
@@ -1204,8 +1279,9 @@ def addBookmarkDict(
     def add_outline_item(
         self,
         title: str,
-        pagenum: int,
+        pagenum: Union[None, PageObject, IndirectObject, int],
         parent: Union[None, TreeObject, IndirectObject] = None,
+        before: Union[None, TreeObject, IndirectObject] = None,
```

Comment:
This is a breaking change as it changes the order of parameters. Would it be possible to move the new parameter to the end?

Reply:
not really : there is a check on the type of italic that allows to detect weither "before" parameter is present or not: the new function is compatible with previous arguments.

Reply:
@MartinThoma OK for you ?